_posts/2023-11-09-in-progress-wildfly-feature-on-openshift.adoc

```diff
@@ -0,0 +1,297 @@
+
+* *Capability config:*
+
+** *Authentication flow: Standard flow, Direct access grants*
+
+* *Login Settings:* leave it blank for now
+
+. Navigate to the *Client Scopes* tab for *myclient*. Here you will see that some scope values are under *Assigned Default Client Scopes* field and some others are under *Assigned Optional Client Scopes* field. When client scopes are added to *Default*, the claims associated with them are automatically added to the access token you receive without the need to configure them. Changing it to *Optional* ensures that they are not included by default and you will only get access to claims if you configure the corresponding scope values. This list of scope values you see here are the values that RH-SSO allows and it varies for different OpenID providers. RH-SSO also allows you to create new scopes, but we will not be doing that for this blog. 
+
+. Now, click on *Roles* and create two roles, *user* and *admin*.
+
+. Finally, create a user called *alice* whose *First Name* is *Alice* and *Last Name* is *Smith* and assign her the *user* and *admin* roles. 
+
+. Next, navigate to the `Credentials` tab and create a password for Alice. Toggle the `Temporary` switch off so you are not prompted to update the password after the first login. 
+
+== Build Container Images for WildFly on OpenShift
+
+Normally to use WildFly features that have been released, we just need to add the WildFly Helm release using the chart that is already available on OpenShift. However, since we are making use of a feature that is not released yet, we will need to build our own image of WildFly from the source code and deploy our application on that. 
+
+=== Clone the WildFly and Elytron Projects
+First we need to clone the `WildFly` and the `WildFly-Elytron` projects with the branches that contain the OIDC scope feature: 
+```
+git clone https://github.com/PrarthonaPaul/wildfly.git
+git checkout WFLY-16532
+```
+Next, go back to the directory that contains the `WildFly-Elytron` project and clone the `WildFly-Elytron` project: 
+```
+git clone https://github.com/PrarthonaPaul/wildfly-elytron.git
+git checkout WFLY-16532
+```
+Notice that we are not using `WildFly/WildFly` and the `wildfly-security/wildfly-elytron` repositories. Instead, we are using a forked versions of the projects. 
+However, this feature does not add any changes to the WildFly-core project. So we can just clone the project and stay on the main branch: 
+```
+git clone https://github.com/wildfly/wildfly-core.git
+```
+
+=== Build the WildFly Project
+Next, we need to build these projects to use the local version of the code. To do this, we will specify the SNAPSHOT versions of `Elytron` and `WildFly-core`: 
+```
+cd wildfly-elytron
+mvn clean install 
+
+cd ../wildfly-core
+mvn clean install -Dversion.org.wildfly.elytron=<insert_SNAPSHOT_VERSION_OF_ELYTRON> 
+
+cd ../wildfly
+mvn clean install -Dversion.org.wildfly.core=<insert_SNAPSHOT_VERSION_OF_WILDFLY-CORE>
+```
```

Comment:
oh no, I got  build error for wildfly with these steps:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project wildfly-appclient: Compilation failure
[ERROR] /tmp/wildfly/appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientDependencyProcessor.java:[61,105] cannot find symbol
[ERROR]   symbol:   method listIterator()
[ERROR]   location: interface java.util.Collection<org.jboss.as.server.deployment.module.ModuleDependency>

have you encountered it before and does it happen to you locally with your branches? Maybe the version for wildfly-core is not aligned now

Reply:
I just built the three projects and it seems fine. 
Are you on the correct branch for all 3? 
The ApplicationClientDependencyProcessor.java file and listIterator() method don't seem like they are related to this feature. 
I just rebased the two branches to pull any new changes from upstream in case they were causing any issues.

Reply:
@PrarthonaPaul Thank you, rebasing helped and it builds well now. Maybe there was a problem that a wldfly-core main branch was further than wildfly WFLY-16532 branch previously. I will continue with review toomrrow

Reply:
@PrarthonaPaul in the future, it can happen that the wildfly-elytron/2.x wildfly-core/main branches will merge new features. If your wildfly-elytron will not be rebased to newest 2.x, the build of wildfly-core main branch would fail since some classes might be missing.

So even though there are no changes for your feature in wildfly-core, I suggest adding a new branch for wildfly-core on your fork and using it in this blog, or using a main branch of your wildfly-core fork, but make sure you will not be pushing updates to your main branch. 

This way it will be more bulletproof

Reply:
Sounds good! I will add that. 
Thanks!